Automattic/woocommerce-payments
# Update tracking conditions

#### Changes proposed in this Pull Request





This PR:

- Disables tracking when WooPayments gateway is disabled
- Disables Button location updates when WooPay is disabled






#### Testing instructions




#### Test disabled gateway
1. Add a breakpoint [here](https://github.com/Automattic/woocommerce-payments/blob/6ea513ef9ec64812aa214af8779b376499b74b22/includes/class-woopay-tracker.php#L249) 
2. As a non-admin customer, visit the checkout page. Make sure the breakpoint in step 1 is reached
3. Disable WooPayments from WooCommerce -> Payments
4. As a non-admin customer, visit the checkout page. Make sure the breakpoint in step 1 is not reached.

#### Test button updates event
1. With WooPay enabled, update the options in "Enable WooPay button on selected pages" section 
2. click "Save Changes" 
3. In ~5 minutes check MC Tracks Live view for the `wcadmin_woopay_express_button_locations_updated` event
4. Disable WooPay 
5. Click "Save Changes" in Payments Settings.
6. There should not be an event for `wcadmin_woopay_express_button_locations_updated` from your blog.

-------------------


**Post merge**
